mobile_verifier/src/data_session.rs

```diff
@@ -0,0 +1,155 @@
+use chrono::{DateTime, Utc};
+use file_store::{file_info_poller::FileInfoStream, mobile_transfer::ValidDataTransferSession};
+use futures::{
+    stream::{Stream, StreamExt, TryStreamExt},
+    TryFutureExt,
+};
+use helium_crypto::PublicKeyBinary;
+use sqlx::{PgPool, Postgres, Transaction};
+use std::collections::HashMap;
+use std::ops::Range;
+use tokio::sync::mpsc::Receiver;
+
+pub struct DataSessionIngestor {
+    pub pool: PgPool,
+}
+
+pub type HotspotMap = HashMap<PublicKeyBinary, u64>;
+
+impl DataSessionIngestor {
+    pub fn new(pool: sqlx::Pool<sqlx::Postgres>) -> Self {
+        Self { pool }
+    }
+
+    pub async fn run(
+        self,
+        mut receiver: Receiver<FileInfoStream<ValidDataTransferSession>>,
+        shutdown: triggered::Listener,
+    ) -> anyhow::Result<()> {
+        tracing::info!("starting DataSessionIngestor");
+        tokio::spawn(async move {
+            loop {
+                tokio::select! {
+                    _ = shutdown.clone() => {
+                        tracing::info!("DataSessionIngestor shutting down");
+                        break;
+                    }
+                    Some(file) = receiver.recv() => self.process_file(file).await?,
+                }
+            }
+
+            Ok(())
+        })
+        .map_err(anyhow::Error::from)
+        .and_then(|result| async move { result })
+        .await
+    }
+
+    async fn process_file(
+        &self,
+        file_info_stream: FileInfoStream<ValidDataTransferSession>,
+    ) -> anyhow::Result<()> {
+        tracing::info!(
+            "handling valid data transfer file {:?}",
```

Comment:
```suggestion
            "handling valid data transfer file {}",
```

It's a string, no reason to use debug fmt here

Reply:
better still
```
tracing::info!(file = file_info_stream.file_info.key, "handling valid data transfer file");
```

Reply:
fixed here: d4ab9d1ee85f20f2779672ba4a806b77614ec2e9